.github/workflows/publish-docs.yaml

```
@@ -0,0 +1,79 @@
+name: publish-docs
+
+on:
+  push:
+    branches:
+      # Branch to base "dev" website on. Set in build.py also.
+      - main
+      # release branches have names like 0.8.x, 0.9.x, ...
+      - "[0-9]+.[0-9]+.x"
+    paths:
+      - "docs/**"
+      - "docsgen/**"
+      - "cli/**"
+      - ".github/workflows/publish-docs.ya?ml"
+      - "Taskfile.ya?ml"
+      - "mkdocs.ya?ml"
+      - "poetry.lock"
+      - "pyproject.toml"
```

```suggestion
      - "pyproject.toml"
      - "go.mod"
      - "go.sum"
```
Since this is relying on the Go dependencies to generate the command line docs, I think it's safest to run the workflow whenever the dependencies change.